### PR TITLE
Fix SAML backchannel SLO failure

### PR DESCRIPTION
**Pull request by GANGANI**
Resolves: https://github.com/wso2/product-is/issues/8160

### CODE REVIEW

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java [hunk 1]
```diff
@@ -930,6 +930,11 @@ public static X509CredentialImpl getX509CredentialImplForTenant(String tenantDom
         X509CredentialImpl credentialImpl = null;
         KeyStore keyStore;
 
+        PrivilegedCarbonContext.startTenantFlow();
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java:933]: shall we use FrameworkUtils.startTenantFlow util method

Comment (GANGANI) [reply components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java]: Fixed https://github.com/wso2-extensions/identity-inbound-auth-saml/pull/303/commits/0602d43ec98400e4afb2d6971692782b5f7c3ab1

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java [hunk 2]
```diff
@@ -931,6 +932,7 @@ public static X509CredentialImpl getX509CredentialImplForTenant(String tenantDom
         KeyStore keyStore;
 
         try {
+            FrameworkUtils.startTenantFlow(tenantDomain);
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java:935]: tenant flow needs to be started and ended only for the tenant case. Can we move the tenant flow starting logic within the
``if (tenantId != -1234) {``` block

A similar fix is done in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1412/files

Comment (GANGANI) [reply components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java:935]: Fixed in https://github.com/wso2-extensions/identity-inbound-auth-saml/pull/307/commits/5935fb2cd0d6b08ffb6efd20003eaa1e0aff90b5

**Pull request merged**